core/lib/server/pub.ts

```diff
@@ -1398,6 +1480,113 @@ export async function getPubsWithRelatedValuesAndChildren<
 						)
 					)
 			)
+			.with("stage_ms", (db) =>
```

Comment:
in these CTEs, we select all the user's stage memberships, all their pub memberships, and any community memberships for the particular community that this query is running in. unlike the query in `lib/capabilities` we don't filter the memberships by anything except the user in these first two CTEs because we don't yet know which pubs or stages are relevant.